addOns/commonlib/src/main/java/org/zaproxy/addon/commonlib/http/ComparableResponse.java

```diff
@@ -468,10 +473,19 @@ private Map<String, Integer> getAllPaths() {
                     }
                 }
                 if (contentType.contains(CONTENT_TYPE_JSON)) {
-                    JSONObject jsonBody = JSONObject.fromObject(body);
-                    for (String childPath : getJsonElementPaths(jsonBody)) {
-                        allPaths.merge(childPath, 1, Integer::sum);
-                        numPaths++;
+                    try {
+                        JSONObject jsonBody = JSONObject.fromObject(body);
+                        for (String childPath : getJsonElementPaths(jsonBody)) {
+                            allPaths.merge(childPath, 1, Integer::sum);
+                            numPaths++;
+                        }
```

Comment:
This should be extracted to avoid duplication.

Reply:
It would be better to extract everything, the for loop as well.

Reply:
There's some side effect here that I'm not seeing/understanding. When it's duplicated things behave when it's extracted they fail (unit test wise).

Reply:
Ok

Reply:
The `getJsonElementPaths` call can be moved there too.

Reply:
Yup I did that a few seconds after I pushed :wink: